File: server/src/game/tools/conf/output/ts/BuildingConfig.ts
class BuildingConfig {
    private static _dic:Dictionary<BuildingConfig> = new Dictionary<BuildingConfig>();
    private static _arr:Array<BuildingConfig> = new Array<BuildingConfig>();

	 public Id :number ;
     public Name :string ;
     public TextId :number ;
     public NameLocation : Array<number>  =new Array<number>();
     public Open :number ;
     public Image :string ;
     public Location : Array<number>  =new Array<number>();
     public GateId :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<BuildingConfig>{
        return  BuildingConfig._dic;
    }
    public static get Arr():Array<BuildingConfig>{
        return BuildingConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:BuildingConfig = <BuildingConfig>json[i];
            BuildingConfig._arr.push(config);
            BuildingConfig._dic.add(config.Id,config);
        }
    }
}

File: tools/conf/output/go/BuildingConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type BuildingConfigElem struct {
	Id int
	Name string
	TextId int
	NameLocation []int
	Open int
	Image string
	Location []int
	GateId int
	
}

type BuildingConfig0 struct {
	Dic map[int]*BuildingConfigElem
	Arr []*BuildingConfigElem
}

var (
	BuildingConfig BuildingConfig0
)

func loadBuildingConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	BuildingConfig.Dic = make(map[int]*BuildingConfigElem)
	BuildingConfig.Arr = make([]*BuildingConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &BuildingConfig.Arr)
	for _,conf := range BuildingConfig.Arr{
		BuildingConfig.Dic[conf.Id] = conf
	}
	
}
